### ISSUE TITLE
Spawning new terminal with workspace folder is broken since v1.62.0-insider

### ISSUE THREAD

**Issue by drweb86** [microsoft/vscode#136099]
Issue Type: <b>Bug</b>

1. Create workspace file 
```
{
    "folders": [
        {
            "name": "scripts",
            "path": "."
        },
        {
            "name": "other",
            "path": ".//other"
        },
    ],
    "settings": {
        "terminal.integrated.cwd": "${workspaceFolder}",
    }
}
```
Create the **other** folder.
2. Open it in Visual Studio Code on Windows
3. Terminal \ New Terminal, select 'scripts'
4. Observe error:
The terminal process failed to launch: Starting directory (cwd) "D:\sources\app\scripts\D:\sources\app\scripts" does not exist.

VS Code version: Code - Insiders 1.62.0-insider (ff1e16eebb93af79fd6d7af1356c4003a120c563, 2021-10-29T05:16:23.014Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 x 2803)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.66GB (21.15GB free)|
|Process Argv|--crash-reporter-id 084d9322-068c-436b-a87b-95494a395812|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by stevenjoezhang** [comment]
I have reproduced a similar issue when setting `"terminal.integrated.cwd": "${fileDirname}"`

**Comment by stevenjoezhang** [comment]
The issue seems to be caused by commit https://github.com/microsoft/vscode/commit/897ed300e6883b5948caf5dc7be7c06a90c55b51
I have replied here https://github.com/microsoft/vscode/commit/897ed300e6883b5948caf5dc7be7c06a90c55b51#commitcomment-59916844

**Comment by drweb86** [comment]
@Tyriar @meganrogge  In the latest insider build this bug it still reproduced with the exact same steps. However folder is different. Please see screenshots below

![image](https://user-images.githubusercontent.com/4712173/147268264-77a39d84-9dcb-409d-9603-1e5f052a462a.png)

![image](https://user-images.githubusercontent.com/4712173/147268289-545aaad0-8a6d-453c-a579-5428dae2bb14.png)

Should we reopen it?

**Comment by Tyriar** [comment]
@drweb86 good point, thanks for raising this. Didn't check the version info hard enough in your report

**Comment by drweb86** [comment]
I want to add some words of appreciation of VS Code. I'm using it for 4 years already and incredibly happy about it. And thinking about lots of efforts in the background that you @Tyriar, @meganrogge and other team members are putting to make this masterpiece editor, I feel envy that I'm not moving the entire world of development forward like you guys...
